### PR TITLE
refactor: Removes deprecation messages for is_slow_operation_thresholding_enabled and unreachable code

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

- [x] Remove deprecation
- [x] Remove unreachable code
- [x] Add a test for reading project resource with an API key with read-only permission

Link to any related issue(s): CLOUDP-280722

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/acceptance-tests.yml [hunk 1]
```diff
@@ -55,7 +55,9 @@ jobs:
     secrets:
       mongodb_atlas_public_key: ${{ inputs.atlas_cloud_env == 'qa' && secrets.MONGODB_ATLAS_PUBLIC_KEY_CLOUD_QA || secrets.MONGODB_ATLAS_PUBLIC_KEY_CLOUD_DEV }}
       mongodb_atlas_private_key: ${{ inputs.atlas_cloud_env == 'qa' && secrets.MONGODB_ATLAS_PRIVATE_KEY_CLOUD_QA || secrets.MONGODB_ATLAS_PRIVATE_KEY_CLOUD_DEV }}
-      mongodb_atlas_gov_public_key: ${{ inputs.atlas_cloud_env == 'qa' && secrets.MONGODB_ATLAS_GOV_PUBLIC_KEY_QA || secrets.MONGODB_ATLAS_GOV_PUBLIC_KEY_DEV }}
+      mongodb_atlas_public_key2: ${{ inputs.atlas_cloud_env == 'qa' && secrets.MONGODB_ATLAS_PUBLIC_KEY2_QA || secrets.MONGODB_ATLAS_PUBLIC_KEY2_DEV }}
```
Comment (EspenAlbert) [inline .github/workflows/acceptance-tests.yml]: using a 2nd API key, to test permission denied error

File: internal/service/project/resource_project_test.go [hunk 2]
```diff
@@ -1080,6 +1080,74 @@ func TestAccProject_withTags(t *testing.T) {
...
+				PreConfig: func() { changeRoles(t, orgID, projectName, "GROUP_READ_ONLY") },
+				Config:    providerConfigPublicKey2 + config,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "is_slow_operation_thresholding_enabled", "false"),
+				),
+			},
+			// Validate the API Key has a different role
+			{
+				Config:      providerConfigPublicKey2 + configBasic(orgID, projectName, "", false, nil, conversion.Pointer(true)),
+				ExpectError: regexp.MustCompile("error in project settings update"),
```
Comment (EspenAlbert) [inline internal/service/project/resource_project_test.go:1111]: Originally, tried to modify existing API key role, but didn't getting an error on update.
Therefore, I had to add another API Key

File: .github/workflows/acceptance-tests-runner.yml [hunk 3]
```diff
@@ -103,6 +103,10 @@ on:
         required: true
       mongodb_atlas_private_key:
         required: true
+      mongodb_atlas_public_key2:
```
Comment (marcosuma) [inline .github/workflows/acceptance-tests-runner.yml]: what about a better naming to indicate its purpose?

**Review by marcosuma — APPROVED** [review]

File: .github/workflows/acceptance-tests-runner.yml (hunk 3 above)
Comment (EspenAlbert) [reply .github/workflows/acceptance-tests-runner.yml]: Not sure. If we use it for other things the generic name might be an advantage.

On the other hand, a more specific name could look like:
`mongodb_atlas_public_key_least_privilege`
or
`mongodb_atlas_public_key_read_only`.
WDYT?

**Review by oarbusi — APPROVED** [review]
LGTM

File: .github/workflows/acceptance-tests-runner.yml (hunk 3 above)
Comment (marcosuma) [reply .github/workflows/acceptance-tests-runner.yml]: `mongodb_atlas_public_key_read_only` is the one I'd vote

Comment (EspenAlbert) [reply .github/workflows/acceptance-tests-runner.yml]: renamed in 4a5352ee3040f80a247f288799a5637dc1fc26c0

**Pull request merged**